app/shared/ast/src/actionCreator/index.ts

```diff
@@ -0,0 +1,1256 @@
+import {
+    ArrowFunctionExpressionNode, getAST,
+    attachCommentsToAst,
+    isArrowFunctionExpression,
+    MemberExpressionNode,
+    isCallExpressionNode,
+    isMemberExpressionNode,
+    LiteralNode,
+    wrapCode,
+    CallExpressionNode,
+    BinaryExpressionNode,
+    BlockStatementNode,
+    isIdentifierNode, isExpressionStatementNode, isTypeOfFunction, 
+} from "../index";
+import {sanitizeScript} from "../utils";
+import {findNodeAt, simple} from "acorn-walk";
+import {Node, Comment} from "acorn";
+import {NodeTypes} from "../constants";
+import {generate} from "astring";
+import {klona} from "klona/json";
+
+const LENGTH_OF_QUOTES = 2;
+const NEXT_POSITION = 1;
+
+export const getTextArgumentAtPosition = (value: string, argNum: number, evaluationVersion: number): string => {
+    // Takes a function string and returns the text argument at argNum position
+    let ast: Node = { end: 0, start: 0, type: "" };
+    let requiredArgument: any = "";
+    let commentArray: Array<Comment> = [];
+    try {
+        // sanitize to remove unnecessary characters which might lead to invalid ast
+        const sanitizedScript = sanitizeScript(value, evaluationVersion);
+        const wrappedCode = wrapCode(sanitizedScript);
+        ast = getAST(wrappedCode, {
+            locations: true,
+            ranges: true,
+            // collect all comments as they are not part of the ast, we will attach them back on line 46
+            onComment: commentArray,
+        });
+    } catch (error) {
+        // if ast is invalid return a blank string
+        return requiredArgument;
+    }
+
+    // attach comments to ast
+    const astWithComments = attachCommentsToAst(ast, commentArray);
```

Comment:
We can remove the comment here as the method name mentions the same information as in comment.
```suggestion
    const astWithComments = attachCommentsToAst(ast, commentArray);
```